test/extensions/filters/http/cache/cache_filter_utils_test.cc

```diff
@@ -0,0 +1,66 @@
+#include "extensions/filters/http/cache/cache_filter_utils.h"
+
+#include "test/test_common/utility.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Cache {
+namespace {
+
+struct IsCacheableRequestParams {
+  Http::TestRequestHeaderMapImpl request_headers;
+  bool is_cacheable;
+};
+
+IsCacheableRequestParams params[] = {
+    {{}, false},
+    {{{":path", "/"}}, false},
+    {{{":path", "/"}, {":method", "GET"}}, false},
+    {{{":path", "/"}, {":method", "GET"}, {"x-forwarded-proto", "https"}}, false},
+    {{{":path", "/"},
+      {":method", "GET"},
+      {"x-forwarded-proto", "https"},
+      {":authority", "test.com"}},
+     true},
+    {{{":path", "/"},
+      {":method", "POST"},
+      {"x-forwarded-proto", "https"},
+      {":authority", "test.com"}},
+     false},
+    {{{":path", "/"},
+      {":method", "GET"},
+      {"x-forwarded-proto", "http"},
+      {":authority", "test.com"}},
+     true},
+    {{{":path", "/"},
+      {":method", "GET"},
+      {"x-forwarded-proto", "http"},
+      {":authority", "test.com"}},
+     true},
+    {{{":path", "/"}, {":method", "GET"}, {"x-forwarded-proto", "ftp"}, {":authority", "test.com"}},
+     false},
+    {{{":path", "/"},
+      {":method", "GET"},
+      {"x-forwarded-proto", "http"},
+      {":authority", "test.com"},
+      {"authorization", "basic YWxhZGRpbjpvcGVuc2VzYW1l"}},
+     false},
+};
+
+class IsCacheableRequestTest : public testing::TestWithParam<IsCacheableRequestParams> {};
+
+INSTANTIATE_TEST_SUITE_P(IsCacheableRequestTest, IsCacheableRequestTest, testing::ValuesIn(params));
+
+TEST_P(IsCacheableRequestTest, IsCacheableRequestTest) {
```

Comment:
I think this way it is easier to add/modify/remove test cases without writing / deleting much code. I tried to make use of whitespace to make the test cases more readable, but then clangd-format just messed it up.

Reply:
I can change them into separate test cases if this will significantly improve readability. I just think it will be more difficult to modify the test cases later. Is there a way to make clangd-format skip modifying whitespaces for a file? Let me know what you think.

Reply:
There is [a way to disable clang-format](http://clang.llvm.org/docs/ClangFormatStyleOptions.html#disabling-formatting-on-a-piece-of-code), but I wouldn't recommend it here--clang-format will prevent more maintenance problems than it causes.

However, I think a good balance would be struck by making separate cacheable and noncacheable TestRequestHeaderMapImpl params, and making separate parameterized tests.

Reply:
That's a great idea. I'm on it.

Reply:
done.